be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/ApiControllers/ProductsApiController.cs

```diff
@@ -79,6 +83,49 @@ public async Task<IActionResult> Get(Guid? categoryId, Guid? brandId, string sea
             return StatusCode((int)HttpStatusCode.OK, products);
         }
 
+        [HttpGet]
+        public async Task<IActionResult> GetProductOrderSuggestion(string searchTerm, bool? hasPrimaryProduct, int pageIndex, int itemsPerPage)
+        {
+            var token = await HttpContext.GetTokenAsync(ApiExtensionsConstants.TokenName);
+            var language = CultureInfo.CurrentUICulture.Name;
+
+            var products = await _productsRepository.GetProductsAsync(
+                token,
+                language,
+                searchTerm,
+                hasPrimaryProduct,
+                GuidHelper.ParseNullable((User.Identity as ClaimsIdentity).Claims.FirstOrDefault(x => x.Type == AccountConstants.Claims.OrganisationIdClaim)?.Value),
+                pageIndex,
+                itemsPerPage,
+                null);
+
+            var onStockProducts = await _inventoryRepository.GetAvailbleProductsInventory(language, pageIndex, itemsPerPage, token);
```

Comment:
**suggestion:** it will be more optimal to get products from the warehouse that come from suggestions - use products.select(x => x.Id);